rrdd/rrdd_server.ml

```diff
@@ -94,6 +94,8 @@ let backup_rrds _ ?(remote_address = None) () : unit =
 					let rrd = Mutex.execute mutex (fun () -> Rrd.copy_rrd rrd) in
 					archive_rrd_internal ~remote_address ~uuid ~rrd ()
 				) vrrds;
+
+			Mutex.lock mutex;
 			let srrds =
```

Comment:
Yes, I agree that it's not the most readable code in general.

There is an additional "simplification": one could use the `try_lock` just to increment the variable, then immediately `unlock` and use the same `Mutex.execute` pattarn for `vrrds`. However it feels to me like wasting a lock in that case.